libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2995,71 +3019,77 @@ bool IRGeneratorForStatements::visit(TryStatement const& _tryStatement)
 	handleCatch(_tryStatement);
-	m_code << "}\n";
+	setLocation(_tryStatement);
+	appendCode() << "}\n";
 
 	return false;
 }
 
 void IRGeneratorForStatements::handleCatch(TryStatement const& _tryStatement)
 {
 	string const runFallback = m_context.newYulVariable();
-	m_code << "let " << runFallback << " := 1\n";
+	appendCode() << "let " << runFallback << " := 1\n";
 
 	// This function returns zero on "short returndata". We have to add a success flag
 	// once we implement custom error codes.
 	if (_tryStatement.errorClause() || _tryStatement.panicClause())
-		m_code << "switch " << m_utils.returnDataSelectorFunction() << "()\n";
+		appendCode() << "switch " << m_utils.returnDataSelectorFunction() << "()\n";
 
 	if (TryCatchClause const* errorClause = _tryStatement.errorClause())
 	{
-		m_code << "case " << selectorFromSignature32("Error(string)") << " {\n";
+		setLocation(*errorClause);
+		appendCode() << "case " << selectorFromSignature32("Error(string)") << " {\n";
 		string const dataVariable = m_context.newYulVariable();
-		m_code << "let " << dataVariable << " := " << m_utils.tryDecodeErrorMessageFunction() << "()\n";
-		m_code << "if " << dataVariable << " {\n";
-		m_code << runFallback << " := 0\n";
+		appendCode() << "let " << dataVariable << " := " << m_utils.tryDecodeErrorMessageFunction() << "()\n";
+		appendCode() << "if " << dataVariable << " {\n";
+		appendCode() << runFallback << " := 0\n";
 		if (errorClause->parameters())
 		{
 			solAssert(errorClause->parameters()->parameters().size() == 1, "");
 			IRVariable const& var = m_context.addLocalVariable(*errorClause->parameters()->parameters().front());
 			define(var) << dataVariable << "\n";
 		}
 		errorClause->accept(*this);
-		m_code << "}\n";
-		m_code << "}\n";
+		setLocation(_tryStatement);
+		appendCode() << "}\n";
+		appendCode() << "}\n";
 	}
 	if (TryCatchClause const* panicClause = _tryStatement.panicClause())
 	{
-		m_code << "case " << selectorFromSignature32("Panic(uint256)") << " {\n";
+		setLocation(*panicClause);
+		appendCode() << "case " << selectorFromSignature32("Panic(uint256)") << " {\n";
 		string const success = m_context.newYulVariable();
 		string const code = m_context.newYulVariable();
-		m_code << "let " << success << ", " << code << " := " << m_utils.tryDecodePanicDataFunction() << "()\n";
-		m_code << "if " << success << " {\n";
-		m_code << runFallback << " := 0\n";
+		appendCode() << "let " << success << ", " << code << " := " << m_utils.tryDecodePanicDataFunction() << "()\n";
+		appendCode() << "if " << success << " {\n";
+		appendCode() << runFallback << " := 0\n";
 		if (panicClause->parameters())
 		{
 			solAssert(panicClause->parameters()->parameters().size() == 1, "");
 			IRVariable const& var = m_context.addLocalVariable(*panicClause->parameters()->parameters().front());
 			define(var) << code << "\n";
 		}
 		panicClause->accept(*this);
-		m_code << "}\n";
-		m_code << "}\n";
+		setLocation(_tryStatement);
```

Comment:
Why not `panicClause`?

Reply:
I must have misunderstood you when you said in the chat

> The the location to the _tryStatement after the clauses.